reanalysis9

```diff
@@ -6,206 +6,216 @@
+total.new = total
+total.new$icd10[grepl("^V|^Z",total.new$icd10)] <- NA
+total.new$icd10_date[grepl("^V|^Z",total.new$icd10)] <- NA
 nrow(total.new)
-#2774909 rows
+#2864652 rows
 length(unique(total.new$eid))
-#395826 individuals
+#498828 individuals
 length(unique(total.new$icd10))
-#1389 ICD codes
-length(unique(total[which(total$mi_date != "NA"), ]$eid))
-#17620 individuals with MI
+#1393 ICD codes
+length(unique(total.new[which(!is.na(total.new$mi_date)), ]$eid))
+#17959 individuals with MI
+
 
 # remove individuals with myocardial infarction outside of timeframe
-total.new = total.new[order(total.new$mi_date), ]
-head(grep('1998-01', total.new$mi_date))
-#43447
-tail(grep('2015-03', total.new$mi_date))
-#269102
-head(grep('NA', total.new$mi_date))
-#272000
-tail(grep('NA', total.new$mi_date))
-#2774909
-total.new = total.new[c(43447:269102, 272000:2774909), ]
-nrow(total.new)
-#2728566 rows
-length(unique(total.new$eid))
-#392743 individuals
-length(unique(total.new$icd10))
-#1389 ICD codes
-length(unique(total[which(total$mi_date != "NA"), ]$eid))
-#17620 individuals with MI
-
-# exclude ICD10 codes outside of timeframe
-total.new.na = total.new[which(total.new$icd10_date == 'NA' | total.new$mi_date == 'NA'), ]
-total.new.temp = total.new[-which(total.new$icd10_date == 'NA' | total.new$mi_date == 'NA'), ]
-total.new.temp$icd10[as.Date(total.new.temp$icd10_date) > as.Date("2015-03-31")] = "NA"
-total.new.temp$icd10[as.Date(total.new.temp$icd10_date) < as.Date("1998-01-01")] = "NA"
-total.new.temp$icd10[as.Date(total.new.temp$mi_date) - as.Date(total.new.temp$icd10_date) < 8] = "NA"
-total.new = rbind(total.new.temp, total.new.na)
-rm(total.new.temp, total.new.na)
+to_remove <- total.new$eid[total.new$mi_date > as.Date("2015-03-01", "%Y-%m-%d") | total.new$mi_date < total.new$startfollowup]
+to_remove <- unique(to_remove[!is.na(to_remove)])
+
+total.new = total.new[!total.new$eid %in% to_remove,]
 nrow(total.new)
-#2728566 rows
+#2816591 rows
 length(unique(total.new$eid))
-#392743 individuals
+#495453 individuals
 length(unique(total.new$icd10))
-#1384 ICD codes
-length(unique(total[which(total$mi_date != "NA"), ]$eid))
-#17620 individuals with MI
-
-# remove individuals with I25
-total.new = total.new[!total.new$icd10 == "I25", ]
+#1393 ICD codes
+length(unique(total.new[which(!is.na(total.new$mi_date)), ]$eid))
+#14584 individuals with MI
+
+# set to missing ICD10 codes/dates outside of timeframe or that are before 7 days from MI date
+total.new$icd10[total.new$icd10_date > as.Date("2015-03-01", "%Y-%m-%d") | total.new$icd10_date < total.new$startfollowup] <- NA
+total.new$icd10_date[total.new$icd10_date > as.Date("2015-03-01", "%Y-%m-%d") | total.new$icd10_date < total.new$startfollowup] <- NA
+total.new$icd10[(total.new$mi_date - total.new$icd10_date) < 8] <- NA
+total.new$icd10_date[(total.new$mi_date - total.new$icd10_date) < 8] <- NA
 nrow(total.new)
```

Comment:
Set to missing all the ICD codes that happen outside the time window (1998-2015) and keep those that are at least 8 days before MI. Double check if this make sense, I didn't think too much about it